R/zchunk_LB162.ag_prodchange_R_C_Y_GLU_irr.R

```diff
@@ -41,67 +45,460 @@ module_aglu_LB162.ag_prodchange_R_C_Y_GLU_irr_DISABLED <- function(command, ...)
+    # For each country, crop, irrigation, the multiplier for each year is
+    # yield in that year / yield in base year.
+
+    # pull off irrigated table of yield, country, crop; append irrigation information
+    FAO_ag_CROSIT %>%
+      select(CROSIT_ctry, CROSIT_crop, year, Yield_kgHa_irrigated) %>%
+      mutate(Irr_Rfd = "IRR") %>%
+      rename(yield_kgHa = Yield_kgHa_irrigated) ->
+      L162.ag_irrYield_kgHa_Rcrs_Ccrs_Y
+
+    # Do same for rainfed and bind the irrigated table.
+    # Then complete missing agricultural years from SPEC_AG_PROD_YEARS and interpolate
+    # to fill in yields. Keep only the years in SPEC_AG_PROD_YEARS.
+    # Finally, Calculate yield multipliers.
+    FAO_ag_CROSIT %>%
+      select(CROSIT_ctry, CROSIT_crop, year, Yield_kgHa_rainfed) %>%
+      mutate(Irr_Rfd = "RFD") %>%
+      rename(yield_kgHa = Yield_kgHa_rainfed) %>%
+      bind_rows(L162.ag_irrYield_kgHa_Rcrs_Ccrs_Y ) %>%
+      # add the missing SPEC_AG_PROD_YEARS and interpolate the yields
+      tidyr::complete(year = c(year, SPEC_AG_PROD_YEARS) ,
+                      CROSIT_ctry, CROSIT_crop, Irr_Rfd) %>%
+      select(CROSIT_ctry, CROSIT_crop, Irr_Rfd, year, yield_kgHa) %>%
+      arrange(year) %>%
+      group_by(CROSIT_ctry, CROSIT_crop, Irr_Rfd) %>%
+      mutate(yield_kgHa = approx_fun(year, yield_kgHa)) %>%
+      ungroup() %>% filter(year %in% SPEC_AG_PROD_YEARS) %>%
+      # calculate yield multipliers
+      group_by(CROSIT_ctry, CROSIT_crop, Irr_Rfd) %>%
+      mutate(Mult = yield_kgHa / first(yield_kgHa)) %>%
+      ungroup() %>%
+      select(-yield_kgHa) %>%
+      # Drop the NaN's = crops with zero base year production / harvested area
+      na.omit() ->
+      L162.ag_Yieldmult_Rcrs_Ccrs_Y_irr
+
+
+    # We apply the above yield multipliers to crop-specific changes in harvested area.
+    # This removes bias from changes in composition of GCAM commodities in the FAO projections.
+    # These yield multipliers are now ready to be matched into the GTAP/LDS-based table of
+    # country x crop x zone harvested area in the base year, L151.ag_irrHA/rfdHA...
     #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
+    # First, merge the separate rainfed and irrigated harvested area datasets to simplify processing.
+    # Then join CROSIT country and crop identifiers. This intermediate table is used in multiple
+    # subsequent pipelines.
     #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+    # CROSIT_ctry identifiers come from the AGLU_ctry table, which needs preprocessing to avoid unwanted
+    # extra rows due to Yugoslav FSR.
+
+    # deal with unfilled Yugoslavia entries in AGLU_ctry without impacting other code chunks.
```

Comment:
Not Ben, but per Page, there are some columns that are fine to update in `AGLU_ctry` because only one chunk uses them.  When I checked in about the CROSIT columns, he indicated they were used by another chunk and I _may_ be better off not risking damaging that. So I erred on the side of caution. 

It's all Yugoslavia, so to me, it would make sense if iso=hrv in Yugoslavia had ALL the same information as iso=hrv in Croatia. But I guess because this issue has been dealt with differently in different chunks at this point, we run the risk of having to go through and tweak each of those.